Tests/MapboxCoreNavigationTests/EHorizonTests.swift

```diff
@@ -0,0 +1,48 @@
+import MapboxNavigationNative
+@testable import MapboxCoreNavigation
+import XCTest
+
+// Happy path repeating EH General Usage Example
+// https://github.com/mapbox/mapbox-navigation-native/blob/0e5ec2b339a129aa67138a4f2cb25a47d909a611/docs/eh_integration.md#general-usage-example
+class EHorizonTests: XCTestCase {
+    var passiveLocationDataSource: PassiveLocationDataSource!
+
+    override func setUp() {
+        super.setUp()
+        passiveLocationDataSource = PassiveLocationDataSource()
+    }
+
+    override func tearDown() {
+        passiveLocationDataSource = nil
+        super.tearDown()
+    }
+
+    func testFreeDriveEHorizon() {
+        try! passiveLocationDataSource.roadObjectsStore.setObserverForOptions(self)
+        passiveLocationDataSource.electronicHorizonDelegate = self
+    }
+
+    var peer: MBXPeerWrapper?
+}
+
+extension EHorizonTests: ElectronicHorizonDelegate {
+    func didUpdatePosition(_ position: ElectronicHorizonPosition, distances: [String : RoadObjectDistanceInfo]) {
+        let graphPosition = try! position.position()
+        _ = try! passiveLocationDataSource.graphAccessor.getEdgeMetadata(forEdgeId: graphPosition.edgeId)
+        _ = try! passiveLocationDataSource.graphAccessor.getEdgeShape(forEdgeId: graphPosition.edgeId)
+    }
+
+    func roadObjectDidEnter(_ objectEnterExitInfo: RoadObjectEnterExitInfo) {}
+
+    func roadObjectDidExit(_ objectEnterExitInfo: RoadObjectEnterExitInfo) {}
+}
+
+extension EHorizonTests: RoadObjectsStoreObserver {
+    func onRoadObjectAdded(forId id: String) {
+        _ = try! passiveLocationDataSource.roadObjectsStore.getRoadObjectMetadata(forRoadObjectId: id)
```

Comment:
Can you add comments explaining what this test is trying to exercise? At a glance, it’s a bit confusing to see a test in this repository exercise only MapboxNavigationNative APIs.

Reply:
I added it as a reference to the generic EH use-case. The goal was to substitute all the APIs with Swift wrappers and be able to remove NavNative import. That can be done now. 
Eventually, this test isn't testing any logic, so it could be removed if no other application is found.